tests/e2e/specs/modules/tagmanager/setup.test.js

```diff
@@ -169,4 +177,40 @@ describe( 'Tag Manager module setup', () => {
 		await expect( page ).toMatchElement( '.googlesitekit-setup-module--tag-manager .mdc-button', { text: /create an account/i } );
 		await expect( page ).toMatchElement( '.googlesitekit-setup-module--tag-manager .googlesitekit-cta-link', { text: /re-fetch my account/i } );
 	} );
+
+	describe.each(
+		Object.keys( allowedAmpModes )
+	)( 'AMP mode %s', ( mode ) => {
```

Comment:
Let's avoid using `.each` in this way. It results in less code, but is less clear to read. We can use utility functions to reduce repetition if needed, but I would prefer to see more precise tests here than less code.

We can leverage nesting as well to reduce duplication as well:
```js
describe( 'Setup with AMP active', () => {
	beforeEach( async () => {
		await activatePlugin( 'amp' );
		await activatePlugin( 'e2e-tests-module-setup-tagmanager-api-mock' );
	} );
	afterEach( async () => {
		await deactivatePlugin( 'amp' );
	} );
	describe( 'with Primary AMP', () => {
		//...
	} )
	describe( 'with Secondary AMP', () => {
		//...
	} )
} )
```

Note that "primary" and "secondary" are more relevant language in terms of integration with Site Kit than the actual name of the mode shown in the AMP plugin.

Reply:
updated.